mlflow/models/evaluation/default_evaluator.py

```diff
@@ -463,21 +478,68 @@ def _evaluate_custom_metric(custom_metric_tuple, eval_df, builtin_metrics):
     :param custom_metric_tuple: Containing a user provided function and its index in the
                                 ``custom_metrics`` parameter of ``mlflow.evaluate``
     :param eval_df: A Pandas dataframe object containing a prediction and a target column.
-    :param builtin_metrics: A dictionary of metrics produced by the default evaluator.
-    :return: A scalar metric value.
+    :param builtin_metrics: A dictionary of aggregate metrics produced by the default evaluator.
+    :param metrics: A dictionary of metric values produced by the default evaluator.
+    :return: MetricValue
```

Comment:
Since metrics and builtin_metrics both are metrics one being `float` and other being `MetricValue` can we just reuse the `builtin_metrics` here anyway?

We are anyway checking for type below to distinguish which one use and also checking if the value is numeric or not below. Then is there is a reason we need both of these variables?